src/test/java/com/tradeshift/HelloWorldResourceTest.java

```diff
@@ -0,0 +1,83 @@
+package com.tradeshift;
+
+
+import com.tradeshift.model.DBMessage;
+import com.tradeshift.model.Message;
+import com.tradeshift.resources.HelloWorldResource;
+import com.tradeshift.service.HelloWorldService;
+import com.tradeshift.service.HelloWorldServiceImpl;
+import com.tradeshift.service.MessagesDAO;
+import junit.framework.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.sql.Timestamp;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.*;
+
+public class HelloWorldResourceTest {
+
+    private MessagesDAO mockMessagesDAO;
+    private HelloWorldService mockHelloWorldService;
+    private HelloWorldResource helloWorldResource;
+
+    @Before
+    public void SetUp() {
+        this.mockMessagesDAO = mock(MessagesDAO.class);
+        this.mockHelloWorldService = mock(HelloWorldService.class);
+        this.helloWorldResource = new HelloWorldResource(mockHelloWorldService);
+
+        when(this.mockHelloWorldService.getHelloWorldMessage("Taso")).thenReturn("hello Taso");
+    }
+
+    @Test
+    public void test_returnTenResults() {
+
+        MessagesDAO mockMessagesDAO = mock(MessagesDAO.class);
+        List<DBMessage> mockMessagesList = new ArrayList<>();
+
+        for (int i=0; i< 10; i++) {
+            mockMessagesList.add(new DBMessage("Taso",new Timestamp(0)));
+        }
+        when(mockMessagesDAO.getMessages(10)).thenReturn(mockMessagesList);
+
+        HelloWorldService helloWorldService = new HelloWorldServiceImpl(mockMessagesDAO);
```

Comment:
Since this is a resource test, you shouldn't test the service. The service should be tested in a separate test and ONLY the functionality that is in the resource would be tested here. In general, try to keep the "unit under test" as small as possible.